effect/testing.py

```diff
@@ -164,3 +164,69 @@ def resolve_stubs(dispatcher, effect):
             break
 
     return effect
+
+
+class EQDispatcher(object):
+    """
+    An equality-based (constant) dispatcher.
+
+    This dispatcher looks up intents by equality and performs them by returning
+    an associated constant value.
+
+    Users provide a mapping of intents to results, where the intents are
+    matched against the intents being performed with a simple equality check
+    (not a type check!).
+
+    e.g.::
+
+        >>> sync_perform(EQDispatcher({MyIntent(1, 2): 'the-result'}),
+        ...              Effect(MyIntent(1, 2)))
+        'the-result'
+
+    assuming MyIntent supports ``__eq__`` by value.
+    """
+    def __init__(self, mapping):
+        """
+        :param mapping: Mapping of intents to results.
+        """
+        self.mapping = mapping
+
+    def __call__(self, intent):
+        # Avoid hashing, because a lot of intents aren't hashable.
+        for k, v in self.mapping.items():
+            if k == intent:
+                return sync_performer(lambda d, i: v)
+
+
+class EQFDispatcher(object):
+    """
+    An Equality-based function dispatcher.
+
+    This dispatcher looks up intents by equality and performs them by invoking
+    an associated function.
```

Comment:
Might want to document that the function needs to take a single argument: the intent